### PR TITLE
allow queries on DB-Servers

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Allow queries on DB-Servers again, for debugging purposes.
This is not an official supported feature, but is sometimes used for debugging. Previous changes made it impossible to run a query on a local shard.

- [x] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)
- [x] The behavior change can be verified via automatic tests

### Testing & Verification

This PR adds tests that were used to verify all changes:

- [x] Added new **integration tests** (i.e. in shell_client)

http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/10490/

### CODE REVIEW

File: arangod/Aql/RestAqlHandler.cpp [hunk 1]
```diff
@@ -291,6 +291,11 @@ RestStatus RestAqlHandler::useQuery(std::string const& operation, std::string co
       return RestStatus::DONE;
     }
     std::shared_ptr<SharedQueryState> ss = _engine->sharedState();
+    TRI_ASSERT(ss != nullptr);
```
Comment (graetzer) [inline arangod/Aql/RestAqlHandler.cpp:294]: engines always have shared state ?

File: arangod/Aql/RestAqlHandler.cpp [hunk 2]
```diff
@@ -291,6 +291,11 @@ RestStatus RestAqlHandler::useQuery(std::string const& operation, std::string co
...
+    if (ss == nullptr) {
```
Comment (graetzer) [inline arangod/Aql/RestAqlHandler.cpp:295]: check should be unnecessary

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/10495/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/10496/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/10501/

**Comment by jsteemann** [comment]
Tests blue

**Pull request merged**